src/main/java/org/opensrp/service/IdentifierSourceService.java

```diff
@@ -0,0 +1,64 @@
+package org.opensrp.service;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensrp.domain.IdentifierSource;
+import org.opensrp.repository.IdentifierSourceRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class IdentifierSourceService {
+
+	private IdentifierSourceRepository identifierSourceRepository;
+
+	@Autowired
+	public IdentifierSourceService(IdentifierSourceRepository identifierSourceRepository) {
+		this.identifierSourceRepository = identifierSourceRepository;
+	}
+
+	public List<IdentifierSource> findAllIdentifierSources() {
+		return identifierSourceRepository.getAll();
+	}
+
+	public IdentifierSource findByIdentifier(String identifier) {
+		return identifierSourceRepository.findByIdentifier(identifier);
+	}
+	
+	public void addOrUpdate(IdentifierSource identifierSource) {
+		validateFields(identifierSource);
+		if (identifierSource.getId() != null && identifierSource.getId() != 0) {
+			identifierSourceRepository.update(identifierSource);
+		} else {
+			identifierSourceRepository.add(identifierSource);
+		}
+	}
+	
+	private void validateFields(IdentifierSource identifierSource) {
+		if (StringUtils.isBlank(identifierSource.getIdentifier())) {
+			throw new IllegalArgumentException("Identifier value was not specified");
+		}
+
+		if (StringUtils.isBlank(identifierSource.getBaseCharacterSet())) {
```

Comment:
Can this be converted to an if else for readability since an exception is being thrown if the if above